### DISCUSSION TITLE
Invoking services and condition

### DISCUSSION THREAD

**Discussion by Martisek** [statelyai/xstate#1737] [Q&A]
Hello, is there in XState state machine possible to invoke service depending on a specific condition?
I have tried to use following construction, however it seems that it does not work:
```
invoke: [
    {
        id: 'fixPipeline',
        cond: 'pipelineActivateReq',
        src: (context,_) => context.pipilineService.fixPipeline(context.pid),
        onDone: {},
        onError: {}
    },
    {
        id: 'unfixPipeline',
        src: (context,_) => context.pipilineService.unfixPipeline(context.pid),
        onDone: {},
        onError: {}
    } 
]
```

I know that I can solve mentioned problem above with two separate states, but I would like to make a decision only in one state...

Thank you, 
Mira

**Comment by davidkpiano** [comment]
Unfortunately, this is not part of the SCXML spec; you're right, the "correct" thing to do would be to define two separate states, or since you want to do it in one state, two child states in a single compound state:

```js
pipeline: {
  initial: 'needsToBeFixed',
  states: {
    needsToBeFixed: {
      always: [
        { target: 'fixPipeline', cond: 'pipelineActivateReq' },
        { target: 'unfixPipeline' }
      ]
    },
    fixPipeline: {
      invoke: {/* ... */}
    },
    unfixPipeline: {
      invoke: {/* ... */}
    }
  }
});
```

It might seem weird to have a transient `needsToBeFixed` state, but if you think about it like a flowchart, it actually makes sense:

<img width="437" alt="Screen Shot 2020-12-17 at 8 29 23 AM" src="https://user-images.githubusercontent.com/1093738/102494028-0cb2dd80-4042-11eb-9713-2698c06f09db.png">

**Reply by Martisek** [reply]
Hi David,

thank you a lot for your reply. So, I will solve it with two child states in a single compound state, as you mentioned.

M.
